### PR TITLE
[Php81] Handle non-dynamic + dynamic args (non-array, non-scalar) passed to New_ on NewInInitializerRector

### PR DESCRIPTION
**Pull request by samsonasik**
- allow non-dynamic args
- skip dynamic args

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/Php81/Rector/ClassMethod/NewInInitializerRector.php [hunk 1]
```diff
@@ -122,9 +128,76 @@ public function provideMinPhpVersion(): int
...
+                continue;
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    private function isAllowedArray(Array_ $array): bool
```
Comment (TomasVotruba) [inline rules/Php81/Rector/ClassMethod/NewInInitializerRector.php]: This service is getting quite complex. Could you extra 1-2 service that detect if the expression is initializable?

Comment (samsonasik) [reply rules/Php81/Rector/ClassMethod/NewInInitializerRector.php]: I will try.

Comment (samsonasik) [reply rules/Php81/Rector/ClassMethod/NewInInitializerRector.php]: implemented, I created new service `ComplexNewAnalyzer` for it https://github.com/rectorphp/rector-src/pull/1737/commits/626c717437b3a72cf4346912aba3293ca4435827

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**